From 92571aa04d57733ff3026bea66c35895771796d7 Mon Sep 17 00:00:00 2001
From: "chunlong.cao" <chunlong.cao@amlogic.com>
Date: Fri, 9 Jul 2021 19:26:00 +0800
Subject: [PATCH] display:Boot animation rotates at various angles,it display abnormal

PD#SWPL-54196

Problem:
	Boot animation rotates at various angles,it display abnormal

Solution:
	setDisplayProjection for SurfaceComposerClient

Verify:
	on t7_an400 platform

Signed-off-by: chunlong.cao <chunlong.cao@amlogic.com>
Change-Id: Ia157b70ded2c91d58981d2c883e874afb3051ad8
---
 cmds/bootanimation/BootAnimation.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index c5f5a2f7e7a0..5a9f8072ce4c 100755
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -1096,12 +1096,18 @@ status_t BootAnimation::readyToRun() {
     mMaxWidth = android::base::GetIntProperty("ro.surface_flinger.max_graphics_width", 0);
     mMaxHeight = android::base::GetIntProperty("ro.surface_flinger.max_graphics_height", 0);
     ui::Size resolution = displayConfig.resolution;
-    resolution = limitSurfaceSize(resolution.width, resolution.height);
+    //resolution = limitSurfaceSize(resolution.width, resolution.height);
     // create the native surface
+	Rect destRect(resolution.getWidth(), resolution.getHeight());
+	
+	SurfaceComposerClient::Transaction t;
+	t.setDisplayProjection(mDisplayToken, ui::ROTATION_0, destRect, destRect);
+	t.apply();
+	
     sp<SurfaceControl> control = session()->createSurface(String8("BootAnimation"),
             resolution.getWidth(), resolution.getHeight(), PIXEL_FORMAT_RGBA_8888);
 
-    SurfaceComposerClient::Transaction t;
+    //SurfaceComposerClient::Transaction t;
 
     // this guest property specifies multi-display IDs to show the boot animation
     // multiple ids can be set with comma (,) as separator, for example:
-- 
2.28.0

